### PR TITLE
Fix door ownership

### PR DESCRIPTION
**Pull request by sarahsturgeon**
This should fix the wire doors themselves not being owned (or cleaned up / undo'd)

### CODE REVIEW

File: lua/entities/wired_door/init.lua [hunk 1]
```diff
@@ -5,6 +5,20 @@ include('shared.lua')
 
 ENT.WireDebugName = "Door Controller"
 
+local function switch(t)
```
Comment (sarahsturgeon) [inline lua/entities/wired_door/init.lua:8]: moved from being a global at the bottom (i think this is stupid and shouldnt exist but it definitely shouldnt be a global)

File: lua/entities/wired_door/init.lua [hunk 2]
```diff
@@ -203,18 +217,7 @@ function ENT:makedoor(ply,trace,ang,model,open,close,autoclose,closetime,class,h
...
-    if f then
-      if type(f)=="function" then
-        f(x,self)
-      else
-        error("case "..tostring(x).." not a function")
-      end
-    end
-  end
-  return t
+	return entit
```
Comment (sarahsturgeon) [inline lua/entities/wired_door/init.lua:221]: Diff is messed up. The function that creates the door entity now returns the door entity so we can set undo/ownership in the tool

File: lua/weapons/gmod_tool/stools/wire_door.lua [hunk 3]
```diff
@@ -55,18 +55,26 @@ if !ConVarExists("sbox_maxdoors") then CreateConVar("sbox_maxdoors", defaultlimi
...
-		
+
 		local index = ply:UniqueID()
 		Doors[ index ] 			= Doors[ index ] or {}
 		Doors[ index ][1] 	= Doors[ index ][1] or {}
 		table.insert( Doors[ index ][1], entit )
-		
-		
+
+		local door = entit.Entity:makedoor(ply,trace,ang,model,open,close,autoclose,closetime,class,hardware)
```
Comment (sarahsturgeon) [inline lua/weapons/gmod_tool/stools/wire_door.lua:64]: Function now returns the door if it succeeded

File: lua/weapons/gmod_tool/stools/wire_door.lua [hunk 4]
```diff
@@ -55,18 +55,26 @@ if !ConVarExists("sbox_maxdoors") then CreateConVar("sbox_maxdoors", defaultlimi
...
+
 		undo.Create("Door")
 		undo.AddEntity( entit )
+
+		if IsValid( door ) then
+			undo.AddEntity( door )
+			ply:AddCleanup( "doors", door )
+		end
```
Comment (sarahsturgeon) [inline lua/weapons/gmod_tool/stools/wire_door.lua]: If it correctly made the door we add it to the undo (which should set CPPI ownership) and adds it to the same cleanup list

Comment (wrefgtzweve) [reply lua/weapons/gmod_tool/stools/wire_door.lua]: undo is FPP specific iirc, so be careful with that

**Review by wrefgtzweve — APPROVED** [review]

File: lua/weapons/gmod_tool/stools/wire_door.lua (hunk 4 above)
Comment (sarahsturgeon) [reply lua/weapons/gmod_tool/stools/wire_door.lua]: Good call. Both Nadmod and FPP check `cleanup.Add` - I'll use that here instead of `Player:AddCleanup`

**Pull request merged**